pkg/controller/controller_test.go

```diff
@@ -705,6 +708,147 @@ func TestController(t *testing.T) {
 				assert.Equal(t, "specification of the created/updated object does not match the desired spec", resCond.Message)
 			},
 		},
+		"binding secret references are resolved": &testCase{
+			mainClientObjects: []runtime.Object{
+				&core_v1.Secret{
+					ObjectMeta: meta_v1.ObjectMeta{
```

Comment:
I think we also need `TypeMeta`. At least we have it in other tests. It may be that it is no longer needed after my recent refactoring... Separate PR later.

Reply:
There are a bunch of Secrets in this file, and none of them have TypeMeta.